src/lib.rs

```diff
@@ -420,4 +479,59 @@ mod tests {
             deposit + deposit + 1 - 2 * ACCESS_KEY_ALLOWANCE
         );
     }
+
+    #[test]
+    fn test_create_advanced_account() {
+        // Create a new instance of the linkdrop contract
+        let mut contract = LinkDrop::new();
+        // Create the public key to be used in the test
+        let pk: PublicKey = "qSq3LoufLvTCTNGC3LJePMDGrok8dHMQ5A1YD9psbiz"
+            .parse()
+            .unwrap();
+        // Default the deposit to an extremely small amount
+        let deposit = 1_000_000;
+
+        // Create options for the advanced account creation
+        let options: CreateAccountOptions = CreateAccountOptions {
+            full_access_keys: Some(vec![pk.clone()]),
+            limited_access_keys: Some(vec![LimitedAccessKey {
+                public_key: pk.clone(),
+                allowance: U128(100),
+                receiver_id: linkdrop(),
+                method_names: "send".to_string(),
+            }]),
+            contract_bytes: Some(include_bytes!("../target/wasm32-unknown-unknown/release/linkdrop.wasm").to_vec()),
+        };
+
+        // Initialize the mocked blockchain
+        testing_env!(
+            VMContextBuilder::new()
+            .current_account_id(linkdrop())
+            .attached_deposit(deposit)
+            .context.clone()
+        );
+
+        // Create bob's account with the advanced options
+        contract.create_account_advanced(bob(), options);
+    }
```

Comment:
I would test a "view access key" here to ensure expected access keys are added

Reply:
Done in integration tests